.github/actions/lint/action.yml

```diff
@@ -4,15 +4,18 @@ description: Run Linting steps
 runs:
   using: composite
   steps:
+  - name: Initialize and update submodules
+    shell: bash
+    run: |
+      set -e -o pipefail
+      git submodule update --init --recursive
+      
```

Comment:
We only need Flutter in order to lint, so we don't need to fetch all the submodules. So you could move this step down, merge it with the 'Verify Flutter installation' one, and only retrieve Flutter.